packages/statemanager/src/stateManager.ts

```diff
@@ -284,12 +444,49 @@ export class DefaultStateManager extends BaseStateManager implements StateManage
     })
   }
 
+  /**
+   * Adds value to the state trie for the `account`
+   * corresponding to `address` at the provided `key`.
+   * @param address -  Address to set a storage value for
+   * @param key - Key to set the value at. Must be 32 bytes long.
+   * @param value - Value to set at `key` for account corresponding to `address`.
+   * Cannot be more than 32 bytes. Leading zeros are stripped.
+   * If it is a empty or filled with zeros, deletes the value.
+   */
+  async putContractStorage(address: Address, key: Uint8Array, value: Uint8Array): Promise<void> {
+    if (key.length !== 32) {
+      throw new Error('Storage key must be 32 bytes long')
+    }
+
+    if (value.length > 32) {
+      throw new Error('Storage value cannot be longer than 32 bytes')
+    }
+
+    const account = await this.getAccount(address)
+    if (!account) {
+      throw new Error('putContractStorage() called on non-existing account')
+    }
+
+    value = unpadBytes(value)
+    if (!this._storageCacheSettings.deactivate) {
+      const encodedValue = RLP.encode(value)
+      this._storageCache!.put(address, key, encodedValue)
+    } else {
+      await this._writeContractStorage(address, account, key, value)
+    }
+  }
+
   /**
    * Clears all storage entries for the account corresponding to `address`.
    * @param address -  Address to clear the storage of
    */
   async clearContractStorage(address: Address): Promise<void> {
-    await this._modifyContractStorage(address, (storageTrie, done) => {
+    const account = await this.getAccount(address)
+    if (!account) {
+      throw new Error(`clearContractStorage() called on non-existing account (${address})`)
```

Comment:
If you selfdestruct when you create a contract, would this not throw?

Reply:
Feeling a bit shaky here, also can vaguely remember that I battled a bit with this addition here (can't fully recall the context though).

I followed the `SELFDESTRUCT` call chain again, seems reasonable to me that a SELFDESTRUCT can also be executed on a non-existing account and at the end we get into this situation, so I updated here: https://github.com/ethereumjs/ethereumjs-monorepo/pull/2634/commits/b13f85b04a625298301236caaac9f2581fb43eb6

I also ran EVM and VM (API) tests which passed.